llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -441,6 +446,94 @@ void createUsedAspectsMetadataForFunctions(
   }
 }
 
+void createConditionallyUsedAspectsMetadataForFunctions(
+    FunctionToConditionExpressionsAndConditionalAspectsTy
+        &FunctionToConditionExpressionsAndConditionalAspects,
+    const AspectsSetTy &ExcludeAspectVals) {
+  for (auto &[F, ConditionsAndAspectsVec] :
+       FunctionToConditionExpressionsAndConditionalAspects) {
+    LLVMContext &C = F->getContext();
+
+    // Create a set of unique conditions and aspects. First we add the ones from
+    // the found aspects that have not been excluded.
+    ConditionExpressionsAndConditionalAspectsTy UniqueConditionsAndAspects;
+    for (const auto &ConditionsAndAspectsPair : ConditionsAndAspectsVec) {
+      AspectsSetTy FilteredAspects;
+      for (const auto &A : ConditionsAndAspectsPair.second)
+        if (!ExcludeAspectVals.contains(A)) {
+          FilteredAspects.insert(A);
+        }
```

Comment:
```suggestion
        if (!ExcludeAspectVals.contains(A))
          FilteredAspects.insert(A);
```

However, I think that coding standard is fine with adding brackets to `for` loop if you still want them here. You can also use [`llvm::copy_if`](https://llvm.org/doxygen/namespacellvm.html#a4d13fd05401d8eb0c97b9864a0eb6028)

Reply:
Didn't add `llvm::copy_if` as `std::inserter` doesn't work with sets